### PR TITLE
Allow custom types as indexes when joining tables

### PR DESCRIPTION
**Pull request by emarj**
See #198 

The change is quite simple. We check if the field is tagged as primary key, if so, we add it as an index regardless of its type.

### CODE REVIEW

**Comment by emarj** [comment]
The change is quite simple. We check if the field is tagged as primary key, if so, we add it as an index, regardless of its type.

**Comment by go-jet** [comment]
Good catch. :100:  Could you also include one test in `./tests/postgres` (or mysql).

File: qrm/scan_context.go [hunk 1]
```diff
@@ -187,28 +187,30 @@ func (s *ScanContext) getGroupKeyInfo(
 		field := structType.Field(i)
 		fieldType := indirectType(field.Type)
 
-		if !isSimpleModelType(fieldType) {
-			if fieldType.Kind() != reflect.Struct {
-				continue
-			}
+		if !isPrimaryKey(field, primaryKeyOverwrites) {
+			if !isSimpleModelType(fieldType) {
+				if fieldType.Kind() != reflect.Struct {
```
Comment (go-jet) [inline qrm/scan_context.go:192]: Maybe we can reorder this condition, to avoid two ifs with negations. 
```golang
if isPrimaryKey(...) {
...
} else !isSimpleModelType(fieldType) {
...
}
```

Comment (go-jet) [reply qrm/scan_context.go:192]: Actually it can be just:

```golang
if isPrimaryKey(...) {
...
} else fieldType.Kind() == reflect.Struct {
...
}
```

Comment (emarj) [reply qrm/scan_context.go]: Yeas you are right. Done!

**Comment by emarj** [comment]
I added the test. Unfortunately I was not able to properly configure the test environment on my local machine. I hope it passes on CI.

File: tests/postgres/select_test.go [hunk 2]
```diff
@@ -658,6 +660,98 @@ ORDER BY city.city_id, address.address_id, customer.customer_id;
...
+WHERE (city.city = 'London'::text) OR (city.city = 'York'::text)
+ORDER BY city.city_id, address.address_id, customer.customer_id;
+`, "London", "York")
+
+	err := stmt.Query(db, &dest)
+
+	require.NoError(t, err)
+	require.Equal(t, len(dest), 2)
+	testutils.AssertJSON(t, dest, `
+[
```
Comment (go-jet) [inline tests/postgres/select_test.go:714]: It seems that tests fails because, json doesn't match. Expected json:

```json
[
	{
		"CityID": {
			"Int64": 312,
			"Valid": true
		},
		"CityName": "London",
		"Customers": [
			{
				"CustomerID": {
					"Int64": 252,
					"Valid": true
				},
				"LastName": "Hoffman",
				"Address": {
					"AddressID": {
						"Int64": 256,
						"Valid": true
					},
					"AddressLine": "1497 Yuzhou Drive"
				}
			},
			{
				"CustomerID": {
					"Int64": 512,
					"Valid": true
				},
				"LastName": "Vines",
				"Address": {
					"AddressID": {
						"Int64": 517,
						"Valid": true
					},
					"AddressLine": "548 Uruapan Street"
				}
			}
		]
	},
	{
		"CityID": {
			"Int64": 589,
			"Valid": true
		},
		"CityName": "York",
		"Customers": [
			{
				"CustomerID": {
					"Int64": 497,
					"Valid": true
				},
				"LastName": "Sledge",
				"Address": {
					"AddressID": {
						"Int64": 502,
						"Valid": true
					},
					"AddressLine": "1515 Korla Way"
				}
			}
		]
	}
]
```

Comment (emarj) [reply tests/postgres/select_test.go:715]: Yes, it was silly on my part. I am so used to `null.Int` by `github.com/guregu/null` that I forgot about the JSON marshaling of `sql.NullInt64`. I also added another test by using the aforementioned `null.Int`.

I was able to configure the environment to run some test, even tough not everything works. I had to pass trough some hoops to make it work on my M1 Mac. In my case `platform: linux/amd64` was needed in `docker-compose.yaml`.

Comment (go-jet) [reply tests/postgres/select_test.go:715]: Yeah, I've heard M1 Mac tends to have problems with docker. Maybe we can include your `docker-compose.yaml` changes. M1 is getting popular, someone else might stumble on the same issue. It can be a separate PR.

**Comment by go-jet** [comment]
Perfect. LGTM.
